src/Native/Runtime/CachedInterfaceDispatch.cpp

```diff
@@ -86,7 +86,7 @@ static void * UpdatePointerPairAtomically(void * pPairLocation,
                                           void * pSecondPointer,
                                           bool fFailOnNonNull)
 {
-#if defined(_X86_) || defined(_ARM_)
+#if defined(_X86_) || defined(_ARM_) || defined(_WASM_)
```

Comment:
Should this ifdef rather be changed to BIT32 ?

Reply:
It looks like BIT32 is managed-only

Reply:
It is meant to be defined for native on 32-bit platforms as well: https://github.com/dotnet/corert/blob/master/src/Native/CMakeLists.txt#L32 . It is just does not seem to be used anywhere yet.

Or you can flip this around and use BIT64 here. There is a lot of uses of BIT64.

Reply:
It looks like BIT32 is only defined for ARM, so I used !BIT64 here.